client/src/hooks/common/useDebouncedFormValue.ts

```diff
@@ -14,8 +14,15 @@ export const useDebouncedFormValue = <T>(
     handleEagerValueChange(eagerValue);
   }, [eagerValue]);
 
+  const parseSpecialChar = (value: any) =>{
```

Comment:
This is a general-purpose hook, let's move this code to `useGetRegexMatches`

Reply:
It will not work during input, only after opening New category matcher modal

Reply:
Yes, but I think that's fine though. The biggest annoyance with this bug is that once you opened `New category matcher modal`, you initially saw no matching transactions at all. 

The input is meant to accept regex, so it should be initially filled with properly escaped special characters.